openlibrary/components/IdentifiersInput.vue

```diff
@@ -115,40 +115,48 @@ export default {
     },
 
     computed: {
-        popularEditionConfigs: function() {
-            if (this.edition_popular) {
-                const popularConfigs = JSON.parse(decodeURIComponent(this.edition_popular));
-                return Object.fromEntries(popularConfigs.map(e => [e.name, e]));
+        parsedConfigs: function() {
+            return JSON.parse(decodeURIComponent(this.id_config_string))
+        },
+        popularIdDict: function() {
+            if (this.output_selector.includes('Edition')) {
+                const popularConfigs = this.parsedConfigs.filter((config) => {
+                    if (popularEditionIds.includes(config.name)) {
+                        return true;
+                    }
+                    return false;
+                })
+                return Object.fromEntries(popularConfigs.map(e => [e.name, e]))
             }
             return {};
         },
-        secondaryEditionConfigs: function() {
-            if (this.secondary_identifiers) {
-                const secondConfigs = JSON.parse(decodeURIComponent(this.secondary_identifiers));
-                return Object.fromEntries(secondConfigs.map(e => [e.name, e]));
+        secondaryIdDict: function() {
+            if (this.output_selector.includes('Edition')) {
+                const filteredConfigs = this.parsedConfigs.filter((config) => {
+                    if (popularEditionIds.includes(config.name)) {
+                        return false;
+                    }
+                    return true
+                })
+                return Object.fromEntries(filteredConfigs.map(e => [e.name, e]))
             }
             return {};
         },
-        identifierConfigsByKey: function(){
-            const parsedConfigs = JSON.parse(decodeURIComponent(this.id_config_string));
-            if (this.isEdition) {
-                const betterConfigs = JSON.parse(decodeURIComponent(this.id_config_string));
-                return Object.fromEntries(betterConfigs.map(e => [e.name, e]));
+        identifierConfigsByKey: function() {
+            if (this.saveIdentifiersAsList) {
+                return Object.fromEntries(this.parsedConfigs.map(e => [e.name, e]));
             }
-            return Object.fromEntries(parsedConfigs['identifiers'].map(e => [e.name, e]));
+            return Object.fromEntries(this.parsedConfigs['identifiers'].map(e => [e.name, e]));
```

Comment:
Can we remove this distinction? It unclear why `parsedConfigs` is sometimes an object, and sometimes a list of strings. Making it always the full objects would be cleaner, since they all have a `.identifiers`.